### PR TITLE
Enabled API for encrypted putObject with metadata

### LINKED ISSUE #705: Support for metadata in putObject
One of the requests that we got at the conference here was to support metadata for putObject (just like how we support it in minio-go)

### PR DESCRIPTION
**Pull request by Praveenrajmani**
minio-java already had provision for metadata support for PUT calls. Which were private.

Fixes #705

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -3293,6 +3293,89 @@ public void putObject(String bucketName, String objectName, InputStream stream,
   }
 
 
+  /**
+   * Uploads data from given stream as object to given bucket where the stream size is unknown.
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: Description is wrong to this method.

**Comment by deekoder** [comment]
ping @Praveenrajmani

File: api/src/main/java/io/minio/MinioClient.java (hunk 1 above)
Comment (Praveenrajmani) [reply api/src/main/java/io/minio/MinioClient.java]: description modified :+1:

Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java]: Is this description "Uploads data from given stream as object to given bucket where the stream size is unknown." correct for this version of put object?

Comment (Praveenrajmani) [reply api/src/main/java/io/minio/MinioClient.java]: No .. changed the `unknown` to `known` :+1:

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -3293,6 +3293,92 @@ public void putObject(String bucketName, String objectName, InputStream stream,
   }
 
 
+  /**
+   * Uploads data from given stream as object to given bucket where the stream size is known.
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: Use http://minio.github.io/minio-java/io/minio/MinioClient.html#putObject-java.lang.String-java.lang.String-java.io.InputStream-long-java.util.Map- description as base and modify as per this method

**Pull request merged**